### ISSUE TITLE
Recognize duplicate activities

### ISSUE THREAD

**Issue by laufhannes** [Runalyze/Runalyze#1168]
- [x] Use the `activity_id` (filled for activities from Garmin) to see if an uploaded activity does already exist.
(- [ ] Check if an activity already exists in the time range (start time - end time))

**Comment by mipapo** [comment]
We only have a look at the original time of the activity. We're using activity_id for the regonization. (Timestamp of time). 
Yes we thought of adding the distance and duration time of the activity in this check, but that would be overkill.

**Comment by undertrained** [comment]
When opening the add activity form, I get the following error:
```
[error] 26734#0: *44873 FastCGI sent in stderr: "PHP message: PHP Fatal error:  Uncaught exception 'InvalidArgumentException' with message 'Activity id must be numerical.' in /<runalyze_path>/inc/core/Activity/DuplicateFinder.php:49
Stack trace:
#0 /<runalyze_path>/inc/training/class.TrainingFormular.php(92): Runalyze\Activity\DuplicateFinder->checkForDuplicate(NULL)
#1 /<runalyze_path>/inc/html/formular/class.Formular.php(163): TrainingFormular->prepareForDisplayInSublcass()
#2 /<runalyze_path>/inc/html/formular/class.Formular.php(175): Formular->prepareForDisplay()
#3 /<runalyze_path>/inc/html/formular/class.StandardFormular.php(174): Formular->display()
#4 /<runalyze_path>/inc/import/class.ImporterWindowTabFormular.php(66): StandardFormular->display()
#5 /<runalyze_path>/inc/import/class.ImporterWindowTabFormular.php(48): ImporterWindowTabFormular->displaySingleFormularFor(Object(TrainingObject))
#6 /<runalyze_path>/inc/import/class.ImporterWindowTab.php(55): ImporterWindowTabFormular->displayTab()
#7 /<runalyze_path>/inc/import/class.ImporterWindow" while reading upstream, client: <clientip>, server: <servername>, request: "GET /call/call.Training.create.php HTTP/1.1", <...>"
```
Proposed fix is in branch `bugfix/master`.

**Comment by laufhannes** [comment]
Fix is merged as 0ecf6c9d497d86bc3300d37b4e8082a64b7d6928.

Still, I'm not sure if that fix is enough. An existing data object can contain `null` (or maybe an empty string `''` as well if the class does not handle that case correctly).

**Comment by undertrained** [comment]
Yeah, the fix contained a typo (and therefore always skipped the duplicate check). `dataObject` is also always set in my (limited) testing. Although the fix was wrong, at least I could add my missing activity ;)
Maybe we should explicity check for `!is_null($this->dataObject->get('activity_id'))`? Anyhow, unless we uploaded a file (garmin), the check should not be run I suppose..
